### ISSUE TITLE
Frozen Highlighting

### ISSUE THREAD

**Issue by trugwaldsaenger** [hbz/oerworldmap#323]
If i click on "the saylor foundation" when coming from the OER Research Hub Story a map of the US East coast is shown: 
https://oerworldmap.org/resource/urn:uuid:53dbfe26-26fd-11e5-b5c4-7f1fe654f2b0
On this map appear two issues:
a) The Saylor Foundation Pin is not highlighted
b) When mouseover the shown pins the highlighting frozens so that all pins become highlightend.
![unbenannt](https://cloud.githubusercontent.com/assets/6639965/9269594/7883e840-426b-11e5-8ecf-dea3749f6ed4.JPG)

**Comment by literarymachine** [comment]
We originally had only the pin for the organization on the map, which was not highlighted. It is like this again, now. See beta: https://beta.oerworldmap.org/resource/urn:uuid:53dbfe26-26fd-11e5-b5c4-7f1fe654f2b0

**Comment by acka47** [comment]
+1

**Comment by acka47** [comment]
I take back my "+1". It seems another issue is introduced with this fix. While currently, most of the time an actual story is shown under "Story" in the popup you get when hovering over a pin, with this fix this behaviour is broken.

For example, see the "Saylor Foundation" pin at https://beta.oerworldmap.org/resource/urn:uuid:53dbfe26-26fd-11e5-b5c4-7f1fe654f2b0.

**Comment by literarymachine** [comment]
AFAIK, this was never the case for popups on individual organization pages.
The linked stories are shown only when listing stories on the landing page.
When viewing individual organizations, the stories are listed in the right
column.
Am 14.08.2015 13:45 schrieb "Adrian Pohl" <notifications@github.com>:

**Comment by acka47** [comment]
> AFAIK, this was never the case for popups on individual organization pages.

It acutally is the case right now, just take a look at https://oerworldmap.org/resource/urn:uuid:53dbfe26-26fd-11e5-b5c4-7f1fe654f2b0 where the OER Reaserch Hub Story is shown under story in the "Saylor Foundation" popup.

**Comment by literarymachine** [comment]
Sorry, what I meant was that _before_ the erroneous behavior that led to multiple markers for resources such as [1] being shown, the popup content was different.

Anyhow, I am trying to sanitize the content of the popups. This is not as straightforward as it seems because we don't have data denormalization (#276) in place yet. With 1c7b709 the behavior is now so that

- when viewing an individual resource such as [2], all known related entities and
- when viewing a resource listing such as [3], only those related entities that are also present in the listing

are shown in the popup. This is the best I can come up with right now with the current data structure in ES and without seriously harming performance. It potentially also fixes #326, at least temporarily, until we have typed inverse relations / denormalization in place.

[1]: https://oerworldmap.org/resource/urn:uuid:53dbfe26-26fd-11e5-b5c4-7f1fe654f2b0
[2]: https://beta.oerworldmap.org/resource/urn:uuid:f28009b2-3b5e-11e5-8709-54ee7558c81f
[3]: https://beta.oerworldmap.org/resource/?q=*educa

**Comment by trugwaldsaenger** [comment]
https://beta.oerworldmap.org/resource/urn:uuid:53dbfe26-26fd-11e5-b5c4-7f1fe654f2b0

When doing a mouseover on the sailor foundation pin, the reference to the stories appears not correct: 
1. Instead of "story" it says "article"
2. Instead of the story icon the project icon is displayed.

Additionally Firefox broke down while testing this issue and I cannot open www.oerworldmap.org in Firefox any longer. Firefox is working with other pages, and I can also open www.oerworldmap.org with Chrome and Opera. Any ideas how this can be solved?

**Comment by trugwaldsaenger** [comment]
Good news, Firefox seems to be working again...

**Comment by literarymachine** [comment]
> 1. Instead of "story" it says "article"

Yes. Right now, it simply takes the "internal" type as a label, which is "Article" for "Story" (and "Action" for "Project"). This is the case because with this fix, not only related Stories/Articles are shown, but all related entities. See the popup e.g. at [1].

> 2. Instead of the story icon the project icon is displayed.

No. This is not the project icon, I added a generic "link" item to signal that this is a related entity.

I can put time and effort into fixing this, but I would rather not and move on. We should not forget that although live, all of this is work in progress. Once we have typed inverse links etc in place, the popup are going to be reworked anyways.

> Additionally Firefox broke down while testing this issue and I cannot open www.oerworldmap.org in Firefox any longer.

I am also experiencing some performance-issues with Firefox which might cause it to crash. Will look into this in #329.

[1]: https://beta.oerworldmap.org/resource/urn:uuid:f28009b2-3b5e-11e5-8709-54ee7558c81f
